pkg/controller/nodes/array/handler.go

```diff
@@ -295,6 +295,18 @@ func (a *arrayNodeHandler) Handle(ctx context.Context, nCtx interfaces.NodeExecu
 
 			retryAttempt := subNodeStatus.GetAttempts()
 
+			// fastcache will not emit task events for cache hits. we need to manually detect a
+			// transition to `SUCCEEDED` and add an `ExternalResourceInfo` for it.
+			if cacheStatus == idlcore.CatalogCacheStatus_CACHE_HIT && len(arrayEventRecorder.TaskEvents()) == 0 {
```

Comment:
is there one potential problem here, consider this case

Round 1:
  node 1 - cache-hit
   node 2 - cache-hit
   node 3 - system error
Round 2:
  node 1 - cache hit
  node 2 - cache miss
      - event write failure?

Reply:
Or maybe this will not happen, as when we actually get a system error I think we update the state?
is it possible to simulate this and test this?

One other option is allow admin to move the state from success to running!